lib/9pfs/9pfs.h

```diff
@@ -39,63 +39,150 @@
 
 #include <vfscore/prex.h>
 
-/*
- * Currently supports only the 9P2000.u variant of the protocol.
+/**
+ * enum uk_9pfs_proto - Protocol version;
+ * currently only the 9P2000.u variant of the protocol is supported
  */
 enum uk_9pfs_proto {
+	/*
+	 * Some extensions and minor adjustments are added to
+	 * the basic protocol for Unix systems
+	 */
 	UK_9P_PROTO_2000U,
+	/*
+	 * Some extensions and minor adjustments are added to
+	 * the basic protocol for Linux systems
+	 */
 	UK_9P_PROTO_2000L,
+	/*
+	 * The number of existent protocols;
+	 * in this case, two (UK_9P_PROTO_2000U and UK_9P_PROTO_2000L)
+	 */
 	UK_9P_PROTO_MAX
 };
 
+/**
+ * struct uk_9pfs_mount_data - An entry containing the necessary data
+ * for mounting the filesystem
+ */
 struct uk_9pfs_mount_data {
-	/* 9P device. */
+	/* 9P device */
```

Comment:
Detail what the 9P device is. A directory, an IP address etc.

Reply:
I think this is a device similar to what we have in `vfscore` [here](https://github.com/unikraft/unikraft/blob/staging/lib/vfscore/include/vfscore/mount.h#L52).

Reply:
It seems like it is a structure used to interact with a `virtio` device.

It is initialized in [`uk_9p_dev_connect`](https://github.com/unikraft/unikraft/blob/staging/lib/uk9p/9pdev.c#L237), by calling `dev->ops->connect()`, which resolves to the transport connect function, `mount data->trans->ops->connect()`, which again resolves to [`virtio_9p_connect()`](https://github.com/unikraft/unikraft/blob/staging/plat/drivers/virtio/virtio_9p.c#L73).

The `char *device_identifier` used by `virtio_9p_connect()` is the fs tag (e.g. `fs0`) and is used to connect to [qemu local fs driver](https://wiki.qemu.org/Documentation/9p#9p_Server) (by using `-device virtio-9p-pci [...] -fsdev local).

I think the description should just say that the `*dev` field is used to interact with a `virtio` device, maybe also point to the initialization in `uk_9p_dev_connect()`, anything else I think would be too much.